src/backend/joanie/core/models/products.py

```diff
@@ -435,6 +432,41 @@ def __str__(self):
         active = _("active") if self.is_active else _("inactive")
         return f"[{active}][{self.state}] {self.user} for {self.course_run}"
 
+    @property
+    def grade_cache_key(self):
+        """The cache key used to store enrollment's grade."""
+        return f"grade_{self.uid}"
+
+    @property
+    def is_passed(self):
+        """Get enrollment's grade, store result in cache then return the passed state."""
+        grade = cache.get(self.grade_cache_key)
+
+        if grade is None:
+            try:
+                grade = self.get_grade()
+            except GradeError:
+                return False
+
+        cache.set(
+            self.grade_cache_key, grade, settings.JOANIE_ENROLLMENT_GRADE_CACHE_TTL
+        )
```

Comment:
getting/setting grades to cache should probably be done in the `get_grade` method so that everybody calling it benefits from it?